Mathlib/GroupTheory/Subgroup/Basic.lean

```diff
@@ -2076,6 +2076,19 @@ theorem center_toSubmonoid : (center G).toSubmonoid = Submonoid.center G :=
 #align subgroup.center_to_submonoid Subgroup.center_toSubmonoid
 #align add_subgroup.center_to_add_submonoid AddSubgroup.center_toAddSubmonoid
 
+/-- For a group with zero, the center of the units is the same as the units of the center. -/
+def centerUnitsEquivUnitsCenter (G₀ : Type*) [GroupWithZero G₀] :
+    Subgroup.center (G₀ˣ) ≃* (Submonoid.center G₀)ˣ where
+  toFun := MonoidHom.toHomUnits <|
+    ⟨⟨fun u ↦ ⟨(u : G₀ˣ), fun r ↦ by
+      rcases eq_or_ne r 0 with (rfl | hr)
+      · rw [mul_zero, zero_mul]
+      exact congrArg Units.val <| u.2 <| Units.mk0 r hr⟩, rfl⟩, fun _ _ ↦ rfl⟩
```

Comment:
Anonymous constructors are pretty cursed (in terms of readability) for morphisms now that structures are nested:
```suggestion
    { toFun := fun u ↦ ⟨(u : G₀ˣ), fun r ↦ by
        rcases eq_or_ne r 0 with (rfl | hr)
        · rw [mul_zero, zero_mul]
        exact congrArg Units.val <| u.2 <| Units.mk0 r hr⟩
      map_one' := rfl
      map_mul' := fun _ _ ↦ rfl }
```